### PR TITLE
Grab stack traces when unit test hangs

### PR DESCRIPTION
**Pull request by cole-miller**
We know which binary is hanging now, try to get some backtraces using GDB if it looks like it's hanging.

### CODE REVIEW

**Comment by cole-miller** [comment]
Running the unit test with `--no-fork` exposed a pre-existing minor test issue where we weren't using the `test/lib/sqlite.c` fixture everywhere we should---I just fixed this quickly as well.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -48,20 +48,18 @@ jobs:
...
 
-    - name: OnTestFailure
-      if: ${{ failure() }}
-      uses: actions/upload-artifact@v3
-      with:
-        name: test-suite.log
-        path: test-suite.log
+          ./unit-test --no-fork &
+          pid=$!
+          bash -c "sleep 10m; sudo gdb -p $pid -batch -ex 'thread apply all bt' -ex quit; false" &
```
Comment (letFunny) [inline .github/workflows/build-and-test.yml:61]: Are you sure gdb is in the container image?

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -48,20 +48,18 @@ jobs:
...
         LIBDQLITE_TRACE: 1
       run: |
-          for bin in unit-test integration-test \
-                     raft-core-fuzzy-test raft-core-integration-test \
-                     raft-core-unit-test raft-uv-integration-test \
-                     raft-uv-unit-test
-          do ./$bin || touch any-failed
-          done 2>&1 | tee -a test-suite.log
-          test '!' -e any-failed
+          tests="integration-test \
```
Comment (letFunny) [inline .github/workflows/build-and-test.yml:51]: Can you add a TODO somewhere to revert this once we figure out why it hangs?

Comment (cole-miller) [reply .github/workflows/build-and-test.yml:54]: Done

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (cole-miller) [reply .github/workflows/build-and-test.yml:64]: It is not, good catch! Fixed.

**Pull request merged**